test_bandwidth.py

```diff
@@ -0,0 +1,204 @@
+import os
+import sys
+import time
+import argparse
+from abc import abstractmethod
+
+import torch
+import torch.distributed as dist
+from typing import List
+from prettytable import PrettyTable
+os.environ['CUDA_VISIBLE_DEVICES'] = '1,2,3,4'
+    
+def setup(rank, world_size):
+    os.environ["MASTER_ADDR"] = os.getenv("MASTER_ADDR", "localhost")
+    os.environ["MASTER_PORT"] = os.getenv("MASTER_PORT", "12355")
+    os.environ["RANK"] = os.getenv("RANK", "0")
+    os.environ["WORLD_SIZE"] = os.getenv("WORLD_SIZE", "1")
+    dist.init_process_group(world_size=world_size, rank=rank,
+                            init_method="env://", backend="nccl")
+    torch.cuda.set_device(rank)
+    device_id = torch.cuda.current_device()
+    print("Process running on GPU:", device_id)
+
+"""    
+    BandWidth Computation: https://github.com/NVIDIA/nccl-tests/blob/master/doc/PERFORMANCE.md
+
+    B: Bus bandWidth of the device
+    S: Elements transformed when execute one collective op
+    t: Time cost when execute one collective op 
+    n: Ranks number
+
+    AllReduce:
+        B = S/t * (2*(n-1)/n) = algbw * (2*(n-1)/n)
+    AllGather:
+        B = S/t * (n-1)/n = algbw * (n-1)/n
+    Broadcast:
+        B = S/t = algbw 
+    AllToAll:
+        B = S/t * (n-1)/n = algbw * (n-1)/n
+"""
+
+class CollectiveOps:
+    @abstractmethod
```

Comment:
Just to be precise, if you use `Ops`, it means many operations. I guess you want to use `Op` instaed.